File: OSCommon/org/optimizationservices/oscommon/datastructure/osinstance/Objective.cs
using System;

namespace org.optimizationservices.oscommon.datastructure.osinstance{
	/// <summary>
	/// A data structure class that corresponds to an xml element in the OSiL schema. 
	/// @author Jun Ma	
	/// @version 1.0, 09/01/2005
	/// @since OS 1.0
	/// @copyright (c) 2005
	/// </summary>
	public class Objective{

		public string name = "";
	
		public string maxOrMin = "min";
	
		public int numberOfObjCoef = 0;
		
		public double constant = 0.0;
	
		public double weight = 1.0;
	
		public ObjCoef[] coef = null;

	}//class Objective
}//namespace

File: OSjava/OSCommon/src/org/optimizationservices/oscommon/datastructure/osinstance/Objective.java
/**
 * @(#)Obj 1.0 03/14/2004
 *
 * Copyright (c) 2004
 */
package org.optimizationservices.oscommon.datastructure.osinstance;

/**
 *
 * <p>A data structure class that corresponds to an xml element in the OSiL schema.  
 *  
 * </p>
 * @author Robert Fourer, Jun Ma, Kipp Martin
 * @version 1.0, 03/14/2004
 * @since OS 1.0
*/
public class Objective{
	
	public String name = "";
	
	public String maxOrMin = "min";
	
	public int numberOfObjCoef = 0;
	
	public double constant = 0.0;
	
	public double weight = 1.0;
	
	public ObjCoef[] coef = null;
	
}//class Obj
